src/main/java/io/pravega/connectors/flink/FlinkPravegaWriter.java

```diff
@@ -174,37 +550,47 @@ public void invoke(T event) throws Exception {
                     executorService
             );
         }
-    }
 
-    @Override
-    public void snapshotState(FunctionSnapshotContext context) throws Exception {
-        if (this.writerMode == PravegaWriterMode.ATLEAST_ONCE) {
+        @Override
+        public void close() throws Exception {
+            flushAndVerify();
+            pravegaWriter.close();
+        }
+
+        @Override
+        public List<UUID> snapshotState(long checkpointId, long checkpointTime) throws Exception {
             log.debug("Snapshot triggered, wait for all pending writes to complete");
             flushAndVerify();
+            return new ArrayList<>();
         }
-    }
 
-    @Override
-    public void initializeState(FunctionInitializationContext context) throws Exception {
-        // Nothing to restore.
-    }
+        @Override
+        public void notifyCheckpointComplete(long checkpointId) throws Exception {
+            //do nothing
```

Comment:
Maybe we should throw `UnsupportedOperationException` in these checkpoint methods.   Or are you intending for `StandardWriter` to be used in some checkpoint scenarios?    How practical would it be to have checkpointing enabled but turn off transaction mode?   Maybe WRITE_MODE=AT_LEAST_ONCE should be the determinant, as opposed to isCheckpointEnabled.